### ISSUE TITLE
Add dedicated OpenSearch connector

### ISSUE THREAD

**Issue by ryn9** [trinodb/trino#11377]
Elastic made it latest libraries not work with OpenSearch

Please provide an OpenSearch connector, based off new libraries

Please have a look the blog post about client libraries available for OpenSearch: https://opensearch.org/blog/community/2021/08/community-clients/

**Comment by sondn98** [comment]
Any update on this?

**Comment by akshayjain3450** [comment]
Any update on this?

**Comment by cezar-tech** [comment]
Would be nice to have for my company as well

**Comment by mkhludnev** [comment]
fyi, 
I just run `$ ../../mvnw test -Dtest=TestElasticsearchOpenSearchConnectorTest` on `master` replacining OpenSearch image to `"opensearchproject/opensearch:2.11.1"`
and got an obvious error:
```
[ERROR] io.trino.plugin.elasticsearch.TestElasticsearchOpenSearchConnectorTest --
Time elapsed: 112.9 s <<< ERROR!
ElasticsearchStatusException[Elasticsearch exception [type=illegal_argument_except
ion, reason=Action/metadata line [1] contains an unknown parameter [_type]]]

        Caused by: org.elasticsearch.client.ResponseException: method [POST], host
 [http://localhost:58732], URI [/_bulk?refresh=true&timeout=1m], status line [HTTP
/1.1 400 Bad Request]
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Action/meta
data line [1] contains an unknown parameter [_type]"}],"type":"illegal_argument_ex
ception","reason":"Action/metadata line [1] contains an unknown parameter [_type]"
},"status":400}
```
It makes me think to start with checking Elasticsearch 8.11 for beginning.

**Comment by mkhludnev** [comment]
Does it make sense from 

1. to copy trino-elasticsearch (master) to trino-opensearch, 
2. replace occurrences ES->OS, 
3. copy ElasticsearchContainer from testcontainer, and 
4. bring up OS insecure just for the beginning? 

If it works it let to decouple ES and unblock #20248.  
cc @wendigo

**Comment by hashhar** [comment]
cc: @martint 

@mkhludnev The outline you propose makes sense. That's how we ended up with the MariaDB connector (copied from MySQL + changes).
